src/common.h

```diff
@@ -0,0 +1,257 @@
+
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+
+#include <map>
+#include <string> // Added to see if Mac buildbot is happy
+#include <vector>
+
+namespace libcellml {
+
+/**
+* Vector of base units. NB not const as added to if need be by user-defined units
```

Comment:
Stumbled on this by "luck" and I disagree that this should not be made `const`. My understanding is that this is an API-based list of base units. So, not a user-based list of base units. As a consequence, it should be made `const`. Otherwise, what would happen to that list if someone first uses the API to handle a model `A` that defines some base units, and then use the API to handle a model `B` that doesn't define new base units, but defines a new unit that happens to have the same name as one of the base units in model `A`? IOW, user-based base units should not be tracked in `baseUnitsList`.

(As I have said, I just stumbled on this by "luck", so I may be missing/overlooking something... :))

Reply:
agree that this list of spec-defined base units should not be modifiable by users.

Reply:
Oops, yes, this is a leftover, changed now.